identities/models.py

```diff
@@ -63,13 +63,23 @@ def remove_details(self, user):
             if attribute != "addresses":
                 updated_details[attribute] = "redacted"
             else:
-                # leave addresses in there
-                updated_details[attribute] = value
+                updated_details[attribute] = {}
         self.details = updated_details
         self.communicate_through = None
         self.updated_by = user
         self.save()
 
+    def optout_address(self, scope, address_type=None, address=None):
+        # for each address type (e.g. email, msisdn, etc.)
+        for cur_address_type, addresses in self.details["addresses"].items():
+            if scope == "all" or cur_address_type == address_type:
+                # for each address value (e.g. foo1@bar.com, +27123, etc.)
+                for cur_address, cur_details in addresses.items():
+                    if scope == "all" or cur_address == address:
+                        cur_details["optedout"] = True
+                        self.details["addresses"][cur_address_type][cur_address] = cur_details  # noqa
```

Comment:
See the commit message on https://github.com/praekelt/seed-identity-store/pull/9/commits/24bf60869e4945bfd5228744eb729cf671c771b2